enhancements/machine-api/dedicated-instances.md

```diff
@@ -0,0 +1,129 @@
+---
+title: dedicated-instances
+authors:
+  - "@alexander-demichev"
+reviewers:
+  - "@JoelSpeed"
+  - "@enxebre"
+approvers:
+  - "@JoelSpeed"
+  - "@enxebre"
+creation-date: 2020-09-01
+last-updated: 2020-09-01
+status: provisional
+---
+
+# Dedicated instances
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Make possible for users to create machines which run as dedicated instances. 
+
+## Motivation
+
+Some organizations need to make sure that their workloads are not hosted on the same physical hardware as others.
+
+### Goals
+
+- Provide automation similar to what Machine API supports for spot instances. 
+
+### Non-Goals
+
+- TODO
+
+## Proposal
+
+In order to give users ability to run their workloads on dedicated instances we should do the following things for AWS, GCP and Azure:
+
+- Add ability to enable dedicated instances using Machine's provider spec.
+
+- Validate that provider spec doesn't contain spot instances configuration and dedicated instances at the same time when it's not supported by the cloud provider.
+
+- Make sure that some required changes are supported and implemented on OCP installer side, like proper control plane creation.
```

Comment:
This would read a bit better if it were worded like this I think, dropping the `some` as it makes the sentence less assertive, in a proposal, everything should be assertive
```suggestion
- Make sure that required changes are supported and implemented within the OCP installer (eg. proper control plane creation).
```